src/models/recipe/recipe.types.ts

```diff
@@ -0,0 +1,14 @@
+import { Document, Model } from "mongoose";
+export interface IRecipe {
+  _id: string;
+  name: string;
+  description: string;
+  likes: { [keyof: string]: Date };
+  postedBy: string;
+  images: Array<{ url: string }>;
+  createdAt: Date;
+  updatedAt: Date;
+}
+
+export interface IRecipeDocument extends Document<IRecipe> {}
+export interface IRecipeModel extends Model<IRecipeModel> {}
```

Comment:
hey  👋🏿let me know if you have any questions or issues with this. But the below should work
```suggestion
export interface IRecipeDocument extends IUser, Document {}
export interface IRecipeModel extends Model<IRecipeDocument > {}
```